### PR TITLE
ColorPicker: Right click + drag locks to one axis

### LINKED ISSUE #310: ColorPicker: Right Click+Drag to move thumb exclusively on relative X or Y-axis
### Is your feature request related to a problem? Please describe.
No.

### Describe the solution you'd like
Currently left-clicking and right-clicking +dragging the color selector thumb in the Color Picker does the same thing.
Right-clicking could display an XY axis originating from the currently selected point that allows the thumb to only be moved along one of the axis.

This might be useful when trying to keep a fade at a specific saturation or brightness level, as selecting a color with the same sat/bright levels freely can be tricky.
Also interesting in combination with the functionality from #301 .

### Your setup
Apollo 1.3.0 PreRelease 3

### Additional context

faderightxy

### PR DESCRIPTION
**Pull request by Brendonovich**
Closes #310, allowing for the ColorPicker main thumb to be dragged on either the X or Y axis independently. While doing so, a line representing the axis that is being dragged along is shown. This could be changed to show both the X and Y axes, but I don't believe this would make sense as the thumb would be locked to one, with the other serving no purpose.

### CODE REVIEW

**Comment by mat1jaczyyy** [comment]
image

You should adjust width and height programmatically as well?

**Comment by mat1jaczyyy** [comment]
@rpgaleksy do you think it should also be possible to start right holding anywhere and have it do this? at the moment, it only works if you precisely start dragging the currently selected point

File: Apollo/Components/ColorPicker.xaml [hunk 1]
```diff
@@ -36,6 +36,11 @@
                   </ControlTemplate>
                 </Thumb.Template>
               </Thumb>
+              
+              <Rectangle IsVisible="false" x:Name="TopBar" Fill="#FFFFFF" Width="1" Height="500"/>
+              <Rectangle IsVisible="false" x:Name="BottomBar" Fill="#FFFFFF" Width="1" Height="500"/>
+              <Rectangle IsVisible="false" x:Name="LeftBar" Fill="#FFFFFF" Width="500" Height="1"/>
+              <Rectangle IsVisible="false" x:Name="RightBar" Fill="#FFFFFF" Width="500" Height="1"/>
```
Comment (mat1jaczyyy) [inline Apollo/Components/ColorPicker.xaml]: These Width/Height fall out of range in Pattern editor.

**Comment by rpgaleksy** [comment]
> @rpgaleksy do you think it should also be possible to start right holding anywhere and have it do this? at the moment, it only works if you precisely start dragging the currently selected point

@mat1jaczyyy Don't see a particular reason why, when selecting anywhere you can't really be sure about the exact color value anyway (which is kind of the point of the axes).
If it helps with code consistency and prevents hacky workarounds I think it's fine though.

**Pull request merged**